### PR TITLE
Fix reputation caching

### LINKED ISSUE #218: invalidate reputation cache upon registering
Currently we fail the invalidate the reputation cache, when a participant registers, which leads to the following bug:

* participant registers, and thereby burns his reputation. reputation cache is not invalidated
* participant fails to attend the meetup

-> in the next meetup, the participant (using the app) will fail to register himself because the rpc does still return the old value, whit the `VerifiedUnlinked` status.

### LINKED ISSUE #219: Purging community ceremony does not invalidate the reputation cache
Currently, the reputation cache is per account, so is its cache invalidation., but when we purge a CommunityCeremony, we remove  the reputations for all accounts for it.

We probably need to introduce another cache dirty bit, which is global.

Currently, we fail to invalidate the cache, when a CommunityCeremony is purged, which can lead to a participant trying to register with an outdated reputation.

### LINKED ISSUE #220: Reputation cache fails to update after the initial value has been set
After ceremony 3 as been claimed, all rewards have been paid out, but the reputation rpc fails to return the new reputations.

for reputables, it worked. the new reputation appears: 
```
./target/release/encointer-client-notee -u wss://kusama.api.encointer.org -p 443 reputation 5Eq4njEjmDoW6jKtZ6ngZ82RCcJY8zdRQXPzttfpCQ6X4DWr
3, u0qj92QX9PQ, Reputation::UnverifiedReputable
2, u0qj92QX9PQ, Reputation::VerifiedLinked
```

for bootstrappers ceremony index 3 is missing (irrespective of who called claim_rewards)
```
./target/release/encointer-client-notee -u wss://kusama.api.encointer.org -p 443 reputation 5Cm38saAbnwrxvnGCp4Pqff5K3rKy4awYQnrdqDeHfvW2ZUM
2, u0qj92QX9PQ, Reputation::VerifiedUnlinked
1, u0qj92QX9PQ, Reputation::VerifiedUnlinked
```

for endorsees too:
```
./target/release/encointer-client-notee -u wss://kusama.api.encointer.org -p 443 reputation 5HVq226T1gz9jgPeNK6dkFh9k53wijArsAEVAohzB8uWisTL
```

and newbies too:
```
./target/release/encointer-client-notee -u wss://kusama.api.encointer.org -p 443 reputation 5GByhFCakyintrJjv9zGVL7whBY3F32ChNEkDirxyCATvQLJ
```

quick fix: purge parachain for all rpc noed and resync

### PR DESCRIPTION
**Pull request by pifragile**
* Closes #218 
* Closes #219
* Closes #220

### CODE REVIEW

File: ceremonies/rpc/src/lib.rs [hunk 1]
```diff
@@ -154,24 +163,42 @@ where
...
+		}
+
+		if refresh_cache {
+			self.refresh_reputation_cache(account.clone(), ceremony_info, at);
+		}
+
+		match self.get_storage::<ReputationCacheValue>(cache_key) {
+			Some(reputation_cache_value) => Ok(reputation_cache_value.reputation),
+			None => Err(Error::OffchainStorageNotFound(format!("{:?}", cache_key)).into()),
+		}
```
[the remark this answers is not in the document]
Comment (pifragile) [reply ceremonies/rpc/src/lib.rs]: this is indeed a lot nicer :)

File: ceremonies/src/lib.rs [hunk 2]
```diff
@@ -165,7 +168,14 @@ pub mod pallet {
...
 			debug!(target: LOG, "registered participant: {:?} as {:?}", sender, participant_type);
-			Self::deposit_event(Event::ParticipantRegistered(cid, participant_type, sender));
+			Self::deposit_event(Event::ParticipantRegistered(
+				cid,
+				participant_type,
+				sender.clone(),
+			));
+
+			// invalidate reputation cache
+			sp_io::offchain_index::set(&reputation_cache_dirty_key(&sender), &true.encode());
```
Comment (clangenb) [inline ceremonies/src/lib.rs:178]: If we invalidate the cache before throwing the event, we can spare a clone. :)

**Review by clangenb — COMMENTED** [review]
One comment, looks good otherwise!

**Review by clangenb — APPROVED** [review]
Perfect, looks good to me!

**Pull request merged**